### PR TITLE
Release v0.6.0

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Release summary

Version number: v0.6.0

## Release details

- Makes FetchAICrypto default again
- Bumps web3 dependencies
- Introduces support for arbitrary protocol handling by DM
- Removes custom fields in signing protocol
- Refactors and updates dialogue and dialogues models
- Moves dialogue module to protocols module
- Introduces MultiplexerStatus to collect aggregate connection status
- Moves Address types from mail to common
- Updates FetchAICrypto to work with Agentland
- Fixes circular dependencies in helpers and configurations
- Unifies contract loading with loading mechanism of other packages
- Adds get-multiaddress command to CLI
- Updates helpers scripts
- Introduces MultiInbox to unify internal message handling
- Adds additional linters (eradicate, more pylint options)
- Improves error reporting in libp2p connection
- Replaces all assert statements with proper exceptions
- Adds skill id to envelope context for improved routing
- Refactors IPC pipes
- Refactors core dependencies
- Adds support for multi-page agent configurations
- Adds type field to all package configurations
- Multiple docs updates including additional explanations of contracts usage
- Multiple additional tests and test stability fixes

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `master` branch (left side), from `develop`
- [x] Lint and unit tests pass locally
- [x] I have checked the fingerprint hashes are correct by running (`scripts/generate_ipfs_hashes.py`)
- [x] I have regenerated the latest API docs
- [x] I built the documentation and updated it with the latest changes
- [x] I have added an item in `HISTORY.md` for this release
- [x] I bumped the version number in the `aea/__version__.py` file.
- [x] I bumped the version number in the `docs/version.md` file
- [x] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-deploy`](../deploy-image/README.md#publish))
- [x] I have checked that the documentation about the `aea cli` tool works
- [ ] I have pushed the latest packages to the registry.

## Further comments

-

### CODE REVIEW

File: aea/cli/generate_key.py [hunk 1]
```diff
@@ -57,14 +57,13 @@ def _generate_private_key(type_: str, file: Optional[str] = None) -> None:
...
-    for type_ in types:
-        private_key_file = IDENTIFIER_TO_KEY_FILES[type_] if file is None else file
+    types = list(crypto_registry.supported_ids) if type_ == "all" else [type_]
+    for type__ in types:
+        private_key_file = (
+            PRIVATE_KEY_PATH_SCHEMA.format(type__) if file is None else file
+        )
         if _can_write(private_key_file):
-            create_private_key(type_, private_key_file)
+            create_private_key(type__, private_key_file)
```
Comment (marcofavorito) [inline aea/cli/generate_key.py:66]: why double underscore?

**Review by marcofavorito — APPROVED** [review]
LGTM

File: aea/cli/generate_key.py (hunk 1 above)
Comment (solarw) [reply aea/cli/generate_key.py:66]: `type_` probably  can be named `type_option` or `type_selection`

File: aea/cli/init.py [hunk 2]
```diff
@@ -38,7 +38,9 @@
 @click.option("--reset", is_flag=True, help="To reset the initialization.")
 @click.option("--local", is_flag=True, help="For init AEA locally.")
 @pass_ctx
-def init(ctx: Context, author: str, reset: bool, local: bool):
+def init(  # pylint: disable=unused-argument
```
Comment (solarw) [inline aea/cli/init.py:41]: probably `del unused_parameter`?

File: aea/helpers/search/models.py [hunk 3]
```diff
@@ -360,41 +361,90 @@ def check_validity(self):
         """
         try:
             if self.type == ConstraintTypes.EQUAL:
-                assert isinstance(self.value, (int, float, str, bool))
+                enforce(
```
Comment (solarw) [inline aea/helpers/search/models.py:364]: a possible place for refactoring: to many ifs

**Review by solarw — APPROVED** [review]
LGTM

File: aea/cli/generate_key.py (hunk 1 above)
Comment (DavidMinarsch) [reply aea/cli/generate_key.py:66]: you reuse a variable from outer scope..it really doesn't matter here; just to avoid same var name

**Pull request merged**